dev/reference/tidb-binlog/faq.md

```diff
@@ -1,13 +1,139 @@
 
+## What is the impact of enabling TiDB Binlog on the performance of TiDB?
+
+- There is no impact on the query.
+
+- There is a slight performance impact on `INSERT`, `DELETE` and `UPDATE` transactions. In terms of latency, a p-binlog is written concurrently in the prewrite stage before committing the transactions. Generally, writing binlog is faster than the prewrite, so it does not increase latency. You can check the response time of writing binlog in Pump's monitoring panel.
+
+## What permissions does Drainer's account need to synchronize the downstream TiDB/MySQL?
+
+To synchronize the downstream TiDB/MySQL, Drainer's account requires the following permissions:
+
+* Insert
+* Update
+* Delete
+* Create
+* Drop
+* Alter
+* Execute
+* Index
+* Select
+
+## What can I do if the Pump disk is almost full?
+
+If Pump's GC is normal and the **gc_tso** time in Pump’s monitoring panel is identical with that of the configuration file, you can perform the following steps to reduce the amount of space required for a single Pump:
+
+- Modify the **GC** parameter of Pump to reduce the number of days to retain data.
+
+- Add pump instances.
+
+## What can I do if Drainer synchronization is interrupted?
+
+Execute the following command to check whether the status of Pump is normal and whether all the Pump instances that are not in the `offline` state are running.
+
+{{< copyable "shell-regular" >}}
+
+``` bash
+binlogctl -cmd pumps
+```
+
+Then, check whether the Drainer monitor or log outputs corresponding errors. If so, resolve them accordingly.
+
+## What can I do if Drainer is slow to synchronizes the downstream TiDB/MySQL?
+
+Check the following monitoring items:
+
+- For the **Drainer Event** monitoring metric, check the speed of Drainer synchronizing `INSERT`, `UPDATE` and `DELETE` transactions to the downstream per second.
+
+- For the **SQL Query Time** monitoring metric, check the time Drainer takes to execute SQL statements in the downstream.
+
+Possible causes and solutions for slow synchronization:
+
+- If the synchronized database contains a table without a primary key or unique index, add a primary key to the table.
+
+- If the latency between Drainer and the downstream is high, increase the value of the `worker-count` parameter of Drainer. For cross-datacenter synchronization, it is recommended to deploy Drainer in the downstream.
+
+- If the load in the downstream is not high, increase the value of the `worker-count` parameter of Drainer.
+
+## What can I do if a Pump instance crashes?
+
+If a Pump instance crashes, Drainer cannot synchronize data to the downstream because it cannot obtain the data of this instance. If this Pump instance can return to normal use, Drainer resumes synchronization; if not, perform the following steps:
+
+1. Use [binlogctl to change the state of this Pump instance to `offline`](/dev/reference/tidb-binlog/maintain.md) to discard the data of this Pump instance.
```

Comment:
s/to/will ?

drainer will ignore this pump instance, the operation will not delete any data.

Reply:
Do you suggest replacing "Use binlogctl to change the state of this Pump instance to `offline`" with "Use binlogctl will change the state of this Pump instance to `offline`"? Please make it clear. Thanks~

Reply:
sorry,  I mean s/to discard/will discard/,   but I think keeping `to discard` is ok now...

Reply:
> sorry, I mean s/to discard/will discard/, but I think keeping `to discard` is ok now...

Got it~